integrations/haystack/README.md

```diff
@@ -0,0 +1,301 @@
+``` python3
+from haystack.document_stores import InMemoryDocumentStore
+
+document_store = InMemoryDocumentStore(
+    similarity="cosine",
+    embedding_dim=768,
+    use_gpu=False
+)
+```
+
+Next, create a retriever. The retriever is houses the deep model and is responsible for, given a document or query, generating an embedding such that query embeddings have a high similarity to their relevant document embeddings.
+``` python3
+from haystack.nodes import EmbeddingRetriever
+
+retriever = EmbeddingRetriever(
+    document_store, # pass in document store
+    embedding_model="deepset/sentence_bert",
+    use_gpu=False,
+)
+```
+``` python3
+>>> retriever.embed_queries(["How many protons in a hydrogen atom"])[0]
+array([-0.00331814, -0.16311326, -0.64788855, -0.35724441, -0.26155273,
+       -0.76656055,  0.35976224, -0.6578757 , -0.15693564, -0.1927543 ])
+```
+
+Next, write some files to your document store. These documents can be instances of Haystack's `Document` class or dictionaries containing a `content`. Remember to update the documents' embeddings with `document_store.update_embeddings(retriever)`
+``` python3
+document_store.write_documents([
+    {
+        "title"  : "Looking Glass",
+        "content": "He came on a summer's day "
+                   "Bringin' gifts from far away."
+                   "But he made it clear he couldn't stay."
+                   "No harbor was his home."
+    },
+    {
+        "title"  : "Bobby Darin",
+        "content": "Somewhere beyond the sea."
+                   "Somewhere waiting for me."
+                   "My lover stands on golden sands "
+                   "And watches the ships that go sailin'"
+    }
+])
+document_store.update_embeddings(retriever)
+```
+
+Finally, create a pipeline and run a query using Haystack's `DocumentSearchPipeline`.
+``` python3
+from haystack.pipelines import DocumentSearchPipeline
+
+pipeline = DocumentSearchPipeline(retriever)
+results = pipeline.run(query="Where does my lover stand?", params={"Retriever": {"top_k": 1}})
+print(results)
+```
+```
+{'documents': [<Document: {'content': "Somewhere beyond the sea.Somewhere waiting for me.My lover stands on golden sands And watches the ships that go sailin'", 'content_type': 'text', 'score': 0.6692139119642437, 'meta': {'title': 'Bobby Darin'}, 'embedding': None, 'id': '4714818c608d92edccf5ec8b44fde052'}>], 'root_node': 'Query', 'params': {'Retriever': {'top_k': 1}}, 'query': 'Where does my lover stand?', 'node_id': 'Retriever'}
+```
+
+### Document Retrieval with DeepSparse ###
+To integrate with the DeepSparse Engine, simply replace your Haystack retriever node with an instance of a DeepSparse node.
+``` python3
+from deepsparse.transformers.haystack import DeepSparseEmbeddingRetriever
+
+retriever = DeepSparseEmbeddingRetriever(
+    document_store,
+    model_path="zoo:nlp/masked_language_modeling/bert-base/pytorch/huggingface/bookcorpus_wikitext/3layer_pruned90-none",
```

Comment:
is there a quantized mlm model we can use here? would definitely be faster and smaller

Reply:
we have a pruned, quantized distilbert
@kylesayrs please confirm this works
```suggestion
    model_path="zoo:nlp/masked_language_modeling/distilbert-none/pytorch/huggingface/wikipedia_bookcorpus/pruned80_quant-none-vnni",
```

Reply:
Confirmed to work